### PR TITLE
Ability to use CRI-O as runtime in kube-up (on Ubuntu for now)

### PR DESCRIPTION
**Pull request by saschagrunert**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Analogous to containerd, we also need a job that uses CRI-O, both in
master and node. This PR enables us to install CRI-O automatically
when we use a ubuntu OS base image if it is not present on the image.
This also enables us to inject newer versions of containerd and runc to
help us test against them as well.

**Which issue(s) this PR fixes**:

Refers to https://github.com/kubernetes/kubernetes/pull/87931, https://github.com/kubernetes/test-infra/issues/16589

**Special notes for your reviewer**:

cc @dims @aojea @mrunalp 

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

File: cluster/gce/gci/configure.sh [hunk 1]
```diff
@@ -496,7 +496,31 @@ function install-containerd-ubuntu {
...
+  # Install dependencies
+  apt-get update
+  apt-get install -y curl make socat
+
+  # Install CRI-O
+  curl -fsSL "https://github.com/cri-o/cri-o/releases/download/$UBUNTU_INSTALL_CRIO_VERSION/crio-$UBUNTU_INSTALL_CRIO_VERSION.tar.gz" | tar -xvz
+  make install -C "crio-$UBUNTU_INSTALL_CRIO_VERSION"
+
+  systemctl daemon-reload
+  systemctl start crio
```
Comment (aojea) [inline cluster/gce/gci/configure.sh:523]: does this mean we can have crio and containerd run at the same time?
may this cause a conflict or can both run at the same time?

**Comment by haircommander** [comment]
@saschagrunert, @harche  was also doing work in this space. maybe y'all can work together to get this through?

File: cluster/gce/gci/configure.sh (hunk 1 above)
Comment (saschagrunert) [reply cluster/gce/gci/configure.sh:523]: I think they can run in parallel, yes. Never actually tried it but running docker on CRI-O nodes should work too.

**Comment by harche** [comment]
Hi @saschagrunert, I have a WIP [PR](https://github.com/kubernetes/test-infra/pull/17014) in `test-infra` for adding presubmit job for crio. Considering the [recently merged support for cgroups v2](https://github.com/kubernetes/enhancements/pull/1370), we are hoping to run those tests on Fedora 31+ node. @vpickard is helping me to identify a gcp proejct which can host Fedora 31 image (gcp doesn't have publicly available fedora images).

As of now there are no `presubmit` or `periodic` CRIO tests in `test-infra`.

**Comment by saschagrunert** [comment]
> Hi @saschagrunert, I have a WIP [PR](https://github.com/kubernetes/test-infra/pull/17014) in `test-infra` for adding presubmit job for crio. Considering the [recently merged support for cgroups v2](https://github.com/kubernetes/enhancements/pull/1370), we are hoping to run those tests on Fedora 31+ node. @vpickard is helping me to identify a gcp proejct which can host Fedora 31 image (gcp doesn't have publicly available fedora images).
> 
> As of now there are no `presubmit` or `periodic` CRIO tests in `test-infra`.

Alright is there anything I can do or should we close this one?

**Comment by harche** [comment]
@saschagrunert no need to close this one. 

Maybe we can work together, on adding a crio presubmit job using `ubuntu`. I already have WIP PR for fedora based job, but we can collaborate and co-author that PR to have both Fedora and Ubuntu based `presubmit` jobs for CRIO if you wish. 

Let me know what do you think.

**Comment by saschagrunert** [comment]
Let's close this for now, I think we're moving into a different direction.

**Pull request closed without merging**